### DISCUSSION TITLE
TraceId and the SpanId are Zeros

### DISCUSSION THREAD

**Discussion by vigamage** [open-telemetry/opentelemetry-java-instrumentation#8572] [Q&A] [answered]
I built a very simple Springboot REST API to test the metric collection and tracing. The metric collection is working fine, but I have issues with the tracing.

I am using the opentelemetry javaagent when starting the application.

`java -javaagent:./opentelemetry-javaagent-1.22.0.jar \
     -jar target/spring-tracing-0.0.1-SNAPSHOT.jar \
     --spring.config.location=./src/main/resources/application.properties`

When I call a endpoint exposed from the service, I get below printed in the console.

`2023-05-25 19:15:58.201  INFO [spring-tracing,00000000000000000000000000000000,0000000000000000] 14527 --- [nio-9093-exec-2] c.e.demo.controller.GreetingsController  : Hello 7777
2023-05-25 19:15:58.202 ERROR [spring-tracing,00000000000000000000000000000000,0000000000000000] 14527 --- [nio-9093-exec-2] c.e.demo.controller.GreetingsController  : error log`

As you can see, It prints zeros as the trace id and the span id. 

I know this question is similar to one below. But I have made sure that spring uses opentelemetry sdk version 1.22.0 and i am using the java agent jar for the same version.(1.22.0)

[https://github.com/open-telemetry/opentelemetry-java-instrumentation/discussions/2435](https://github.com/open-telemetry/opentelemetry-java-instrumentation/discussions/2435)

How can I solve this issue.

Thanks

**Comment by trask** [comment, marked as answer]
hi @vigamage, are you following https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/logger-mdc-instrumentation.md?

what happens if you print `Span.current().getSpanContext().getTraceId()` from inside of your code at that point where you are performing the logging?

**Reply by vigamage** [reply]
Actually I was following some other tutorials which talked about trace and span id only. The combination of them together with open telemetry java agent was not a point of discussion of any of them.

Now only I understand that the mdc should come into play here. After adding the MDC to the log lines, it started working. Thanks @trask 

`logging.pattern.level = trace_id=%mdc{trace_id} span_id=%mdc{span_id} trace_flags=%mdc{trace_flags} %5p`

**Comment by banujan6** [comment]
@trask  I get the same issue. When I print the `TraceID` using `Span.current().getSpanContext().getTraceId()`, I get `00000000000000000000000000000000`.

In fact, I build the `Span` using the `Context` from the front-end application.

```
otelSpan = NeutrinoOTel.createSpanFromParent( ( ObjectNode )message.getOtelContext() );
otelScope = otelSpan.makeCurrent();
System.out.println(otelSpan.getSpanContext().getTraceId());
```

The `message.getOtelContext()` is giving the `Context` which is in the request.

```javascript
{
   //... Other request data
    "otel-ctx": {
        traceparent: "00-f4cd9f124f94a8e029c0d5d72078759f-4a4eae6a3402fb14-01"
    }
}
```

**Reply by laurit** [reply]
@banujan6 it is best to create a new discussion for your issue not hijack an old discussion where the issue is completely different from yours. That being said the issue is most likely in your code. Try debugging `NeutrinoOTel.createSpanFromParent`, by stepping   through the code you should be able narrow down why the context isn't extracted. If you need assistance then create a new github issue and include a minimal sample application that reproduces your issue and include all necessary instructions.

**Reply by banujan6** [reply]
This is noted. Thank you!
